payments-core/src/main/java/com/stripe/android/model/PaymentIntent.kt

```diff
@@ -152,13 +152,15 @@ constructor(
 
     override val nextActionData: StripeIntent.NextActionData? = null,
 
-    private val paymentMethodOptionsJsonString: String? = null
-
+    /**
+     * Payment method options for this PaymentIntent.
+     */
+    @get:RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    override val paymentMethodOptionsMap: Map<String, PaymentMethodOptions>? = emptyMap(),
 ) : StripeIntent {
-
-    fun getPaymentMethodOptions() = paymentMethodOptionsJsonString?.let {
-        StripeJsonUtils.jsonObjectToMap(JSONObject(it))
-    } ?: emptyMap()
+    fun getPaymentMethodOptions() = StripeJsonUtils.jsonObjectToMap(
+        paymentMethodOptionsMap?.toJson()
+    ) ?: emptyMap()
```

Comment:
Are we converting this to a JSON only to convert it to a `Map` again? 🤔

Also, we could think about replacing this method with a public `paymentMethodOptions` field.

Reply:
Are we missing keys with that? For example, what about `installments` and `mandate_options`? Since we don’t have them in `PaymentMethodOptions`, will they just not be accessible anymore?

Reply:
Oh good point, you're right. I will add them